cloud/vmware/vmware_vm_shell.py

```diff
@@ -0,0 +1,191 @@
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: vmware_vm_shell
+short_description: Execute a process in VM
+description:
+    - Start a program in a VM without the need for network connection
+version_added: 2.0
+author: "Ritesh Khadgaray (@ritzk)"
+notes:
+    - Tested on vSphere 5.5
+    - Only the first match against vm_id is used, even if there are multiple matches
+requirements:
+    - "python >= 2.6"
+    - PyVmomi
+options:
+    hostname:
+        description:
+            - The hostname or IP address of the vSphere vCenter API server
+        required: True
+    username:
+        description:
+            - The username of the vSphere vCenter
+        required: True
+        aliases: ['user', 'admin']
+    password:
+        description:
+            - The password of the vSphere vCenter
+        required: True
+        aliases: ['pass', 'pwd']
+    datacenter:
+        description:
+            - The datacenter hosting the VM
+        required: False
+    vm_id:
+        description:
+            - The identification for the VM
+        required: True
+    vm_id_type:
+        description:
+            - The identification tag for the VM
+        default: dns_name
+        choices:
+            - 'uuid'
+            - 'dns_name'
+            - 'inventory_path'
+        required: False
+    vm_username:
+        description:
+            - The user to connect to the VM.
+        required: False
+    vm_password:
+        description:
+            - The password used to login to the VM.
+        required: False
+    vm_shell:
+        description:
+            - The absolute path to the program to start. On Linux this is executed via bash.
+        required: True
+    vm_shell_args:
+        description:
+            - The argument to the program. 
+        required: False
+        default: None
+    vm_shell_env:
+        description:
+            - Comma seperated list of envirnoment variable, specified in the guest OS notation
+        required: False
+        default: None
+    vm_shell_cwd:
+        description:
+            - The current working directory of the application from which it will be run
+        required: False
+        default: None
+'''
+
+EXAMPLES = '''
+    - name: shell execution
+      local_action:
+        module: vmware_vm_shell
+        hostname: myVSphere
+        username: myUsername
+        password: mySecret
+        datacenter: myDatacenter
+        vm_id: DNSnameOfVM
+        vm_username: root
+        vm_password: superSecret
+        vm_shell: /bin/echo
+        vm_shell_args: " $var >> myFile "
+        vm_shell_env:
+          - "PATH=/bin"
+          - "var=test"
+        vm_shell_cwd: "/tmp"
+
+'''
+
+try:
+    from pyVmomi import vim, vmodl
+    HAS_PYVMOMI = True
+except ImportError:
+    HAS_PYVMOMI = False
+
+def find_vm(content, vm_id, vm_id_type="dns_name", datacenter=None):
```

Comment:
hi @Lujeni 

  Following the convention of all other modules under cloud/vmware . none of them use finally.